core/src/main/scala/org/broadinstitute/dsde/rawls/bucketMigration/BucketMigrationService.scala

```diff
@@ -27,6 +34,77 @@ class BucketMigrationService(val dataSource: SlickDataSource, val samDAO: SamDAO
     with WorkspaceSupport
     with LazyLogging {
 
+  def getEligibleOrMigratingWorkspaces: Future[Map[String, Option[MultiregionalBucketMigrationProgress]]] = {
+    def getOwnedWorkspaces: Future[Seq[Workspace]] = for {
+      userWorkspaces <- samDAO.listUserResources(SamResourceTypeNames.workspace, ctx)
+      workspaceIds = userWorkspaces
+        .filter(_.hasRole(SamWorkspaceRoles.owner))
```

Comment:
Following up on the discussion in https://github.com/broadinstitute/sam/pull/1183#issuecomment-1721434522… should this be checking for an "own" action instead of an "owner" role? I'm still trying to work through what one should infer outside of Sam with respect to roles.

Reply:
It should be yes, this directly contradicts what I said in that thread by relying on the role name rather than an explicit action.

This API lists the `workspace` resources that the user has access to along with the roles and non-default actions that they have. The `own` action is granted by default in the `owner` role, which means it is not present in this API response. The only way to check if the user has the `own` action is to query each `workspace` resource individually (either by [listing the actions they have](https://sam.dsde-dev.broadinstitute.org/#/Resources/resourceActionsV2) or by [explicitly checking for the `own` action](https://sam.dsde-dev.broadinstitute.org/#/Resources/resourcePermissionV2)). 

This is the lazy and technically incorrect way of doing this permissions check (although relatively safe unless someone removes the `own` action from the `owner` workspace role). I could add the additional authz checks, but I'd say that in this case it's not a huge deal since this is just reading the workspaces and we'll check for the `own` action before actually allowing migration.

Reply:
This code looks at actions.own instead: https://github.com/broadinstitute/rawls/pull/2555/files#diff-cc8ea874ea1b95b1e65e8d09795f295476565b0a44d36b924ad355e659ba61eeR46

Reply:
It does, that's the correct way of doing it. It uses a different Sam API that is per-`workspace` resource. I wanted to do it the right way when possible (which includes all cases where workspaces are actually scheduled for migration) -- it's just expensive in this situation due to the available Sam APIs.